### PR TITLE
Review

### PR DESCRIPTION
**Pull request by Jchinonso**

### CODE REVIEW

File: dist/app.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+app.listen(process.env.PORT, function () {
+	console.log('server now running at ' + process.env.PORT);
+});
+
+// load the routes
+app.use('/', _api2.default);
+
+_superTest2.default.create(app); // load supertest for create endpoint
+_superTest2.default.search(app); // load supertest for search endpoint
```
Comment (EmaEvidence) [inline dist/app.js:36]: I think this should not be uploaded.

**Review by EmaEvidence — COMMENTED** [review]
Good Job but have to adhere to the style guide some of your lines are longer than 80 characters.
dist folder should not be pushed to the repo.

File: src/inverted-index.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        const getTokensPerObject = [];
+        for (const key in fileContent[i]) {
+          const getTokensPerKey = [];
+          const getEachKey = fileContent[i][key].toLowerCase().replace(/\W+/g, ' ').split(' ');
+          for (let j = 0; j < getEachKey.length; j += 1){
+            if (getTokensPerKey.indexOf(getEachKey[j]) === -1){
+              getTokensPerKey.push(getEachKey[j]);
+            }
+          }
+          for (let k = 0; k < getTokensPerKey.length; k += 1) {
```
Comment (Mcdavid95) [inline src/inverted-index.js:87]: you can use more descriptive terms to run loops than using alphabets

File: src/inverted-index.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+ * add it to the object index
+ * @param {string} fileName
+ * @param {Object} fileContent
+ * @return {Object} this.index
+ */
+  createIndex(fileName, fileContent) {
+    if (this.isValidJSON(fileContent) && this.isValidFileName(fileName)) {
+      const innerIndex = {};
+      const getAllUniqueTokens = [];
+      for (let i = 0; i < fileContent.length; i += 1){
```
Comment (Mcdavid95) [inline src/inverted-index.js:77]: you can use more descriptive terms to run loops than using alphabets

File: src/inverted-index.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        }
+        getAllUniqueTokens.push(...getTokensPerObject);
+      }
+      getAllUniqueTokens.sort();
+      getAllUniqueTokens.forEach((term) => {
+        if (innerIndex[term] !== undefined) {
+          innerIndex[term] = [...innerIndex[term], Number(innerIndex[term][innerIndex[term].length - 1] + 1)];
+          }
+        else {
+          for (let l = 0; l < fileContent.length; l += 1){
```
Comment (Mcdavid95) [inline src/inverted-index.js:101]: you can use more descriptive terms to run loops than using alphabets

File: src/inverted-index.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+ * @return {Object} searchResult
+ */
+  searchIndex(index, fileName, ...searchTerms) {
+    const searchResult = {};
+    let getSearchTerms = [];
+
+    if (Object.keys(index).length === 0) {
+      return { error: 'Index has not been created. Kindly create index before searching' };
+    }
+    else {
```
Comment (Mcdavid95) [inline src/inverted-index.js:135]: else statements should start at the terminating curly brace of if statement

**Review by Mcdavid95 — CHANGES_REQUESTED** [review]
Very good logic and directory arrangement

File: dist/app.js (hunk 1 above)
Comment (chinazor-allen) [reply dist/app.js:36]: Your dist folder should be included in your .gitignore file

File: .coverall.yml [hunk 6]
```diff
@@ -1 +1 @@
-repo_token: tHFoKCxd4Ec5EphDy0fQpV833gzIwPkSW
\ No newline at end of file
+repo_token: RwmFjGt4MIaSRMXJgDfDgRIr5ufdu2S82
```
Comment (nelsonabienorin) [inline .coverall.yml:1]: This file contain private data that should git-ignored

File: dist/app.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+'use strict';
```
Comment (nelsonabienorin) [inline dist/app.js:1]: Your dist folder should be git-ignored

File: src/inverted-index.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+class invertedIndex {
+/**
+ * having index as object property
+ * @constructor
+ */
+  constructor() {
+    this.index = {};
+  }
+
+
```
Comment (nelsonabienorin) [inline src/inverted-index.js:17]: Multi-line spaces should be removed

File: src/inverted-index.js [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        getAllUniqueTokens.push(...getTokensPerObject);
+      }
+      getAllUniqueTokens.sort();
+      getAllUniqueTokens.forEach((term) => {
+        if (innerIndex[term] !== undefined) {
+          innerIndex[term] = [...innerIndex[term], Number(innerIndex[term][innerIndex[term].length - 1] + 1)];
+          }
+        else {
+          for (let l = 0; l < fileContent.length; l += 1){
+            for (const keyy in fileContent[l]){
```
Comment (nelsonabienorin) [inline src/inverted-index.js:102]: Any reason for using keyy as a variable name?

File: src/inverted-index.js [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+ * the search looks through the specified file, else it checks
+ * through the whole index object
+ * @param {Object} index : The created index
+ * @param {string} fileName : Name of file to search
+ * @return {Object} searchResult
+ */
+  searchIndex(index, fileName, ...searchTerms) {
+    const searchResult = {};
+    let getSearchTerms = [];
+
```
Comment (nelsonabienorin) [inline src/inverted-index.js:131]: Be on the look out for unnecessary line space

File: src/inverted-index.js [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+        getSearchTerms = getSearchTerms.map((data) => {
+          return data.toLowerCase();
+        });
+        getSearchTerms.sort();
+
+        for (const key in index) {
+          const innerResult = {};
+          getSearchTerms.forEach((val) => {
+            if (index[key][val] !== undefined) {
+              innerResult[val] = index[key][val];
```
Comment (nelsonabienorin) [inline src/inverted-index.js:188]: Can innerResult variable name be changed ?

File: tests/super-test.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import supertest from 'supertest';
+import indexObj from '../src/inverted-index';
+
+const superObj = {
+  create: (app) => {
+         supertest(app)
+        .post('/api/create')
+        .expect('Content-Type', /json/)
+        .expect(200)
+       // .send({ fileName: 'book1.json', fileContent: indexObj.readFile('book1.json')})
```
Comment (nelsonabienorin) [inline tests/super-test.js:10]: Unnecessary comments should be removed

File: tests/inverted-index-test.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
+
+import indexObj from '../src/inverted-index';
+
+
```
Comment (nelsonabienorin) [inline tests/inverted-index-test.js:4]: More test should be written

**Review by nelsonabienorin — COMMENTED** [review]
Kindly effect all these changes where applicable.

**Pull request merged**